task.ps1

```
@@ -1,23 +1,109 @@
-$vmName = "matebox"
-$vmImage = "Ubuntu2204"
-$vmSize = "Standard_B1s"
-
-Write-Host "Creating a resource group $resourceGroupName ..."
-New-AzResourceGroup -Name $resourceGroupName -Location $location
-
-Write-Host "Creating a network security group $networkSecurityGroupName ..."
-$nsgRuleSSH = New-AzNetworkSecurityRuleConfig -Name SSH  -Protocol Tcp -Direction Inbound -Priority 1001 -SourceAddressPrefix * -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 22 -Access Allow;
-$nsgRuleHTTP = New-AzNetworkSecurityRuleConfig -Name HTTP  -Protocol Tcp -Direction Inbound -Priority 1002 -SourceAddressPrefix * -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 8080 -Access Allow;
-New-AzNetworkSecurityGroup -Name $networkSecurityGroupName -ResourceGroupName $resourceGroupName -Location $location -SecurityRules $nsgRuleSSH, $nsgRuleHTTP
-
-# ↓↓↓ Write your code here ↓↓↓
+param()
+
+$ErrorActionPreference = 'Stop'
+
+# ---------- Базові налаштування ----------
+$location          = 'northeurope'
+$resourceGroupName = 'mate-azure-task-9'
+
+$vnetName          = 'vnet'
+$subnetName        = 'default'
+$addressPrefix     = '10.10.0.0/16'
+$subnetPrefix      = '10.10.0.0/24'
+
+$nsgName           = 'defaultnsg'
+
+$publicIpName      = 'linuxboxpip'
+# 👇 ЦЕ ПОТРІБНО ЗМІНИТИ НА ЩОСЬ УНІКАЛЬНЕ (тільки маленькі букви, цифри, '-')
+$dnsLabelPrefix    = 'mate9-maxim'   # приклад: mate9-maxim-2025
+
+$sshKeyName        = 'linuxboxsshkey'
+
+$vmName            = 'matebox'
+$vmSize            = 'Standard_B1s'
+$imageAlias        = 'Ubuntu2204'    # friendly name образу
+
+Write-Host "=== Creating resource group ==="
+New-AzResourceGroup -Name $resourceGroupName -Location $location -Force | Out-Null
+
+Write-Host "=== Creating Network Security Group '$nsgName' ==="
+$null = New-AzNetworkSecurityGroup `
+    -Name $nsgName `
+    -ResourceGroupName $resourceGroupName `
+    -Location $location `
+    -ErrorAction SilentlyContinue
```

Using `-ErrorAction SilentlyContinue` can hide important errors. If this command fails for a reason other than the resource already existing (e.g., a permissions issue or an invalid parameter), the error will be hidden, and the script might fail later at a less obvious point. To make the script safely re-runnable, consider explicitly checking if the resource exists first (e.g., with `Get-AzNetworkSecurityGroup`) before attempting to create it.